content/cloud-computing/concepts/saas/saas.md

```diff
@@ -0,0 +1,38 @@
+---
+Title: 'SaaS'
+Description: 'SaaS is a cloud computing service delivery model that makes software and applications available over the internet.'
+Subjects:
+  - 'Computer Science'
+  - 'Networking'
+Tags:
+  - 'Networking'
+  - 'Cloud'
+  - 'Infrastructure'
+CatalogContent:
+  - 'foundations-of-cloud-computing'
+  - 'back-end-engineer-career-path'
+---
+
+**Software as a service (SaaS)** is a cloud computing service delivery model that makes software and applications available over the Internet.
+
+Some examples of SaaS include Google Workspace, Dropbox, and Microsoft 365.
+
+## Architecture
+
+SaaS services typically use a multi-tenant approach, where multiple customers share access to cloud resources. Each customer has their own space to store data. The data is isolated and will remain invisible to other tenants.
+
+## Advantages
+
+With its flexibility and low costs, SaaS services have several advantages including:
+
+- Accessibility: Requires only an internet connection, can be accessed from an internet-enabled device (e.g., a computer, smartphone or tablet) within a location that has internet service.
+- Management: Customers do not have to maintain anything; all maintenance and updates are provided by the service provider.
+- Scalability: Customers can use as much of the service as they want, and are charged for the amount used.
+
+## Challenges
+
+Since they are not in direct control of the service, customers could face challenges such as:
+
+- Service Issues: Disruptions like security breaches, server outages, or service changes are beyond the customer's control.
+- Switching Providers: Depending on the amount of data being stored, it may be difficult to transfer the data to a different provider.
+- Updates: If the service provider issues a new version of the product, the customer may have to allocate time learning the newer version.
```

Comment:
```suggestion
- **Service Issues**: Disruptions like security breaches, server outages, or service changes are beyond the customer's control.
- **Switching Providers**: Depending on the amount of data being stored, it may be difficult to transfer the data to a different provider.
- **Updates**: If the service provider issues a new version of the product, the customer may have to allocate time learning the newer version.
```